demos/projects/ESPRESSIF/esp32/main/azure_iot_freertos_esp32_main.c

```diff
@@ -0,0 +1,280 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <stdarg.h>
+#include <string.h>
+
+#include "sdkconfig.h"
+#include "esp_event.h"
+#include "esp_wifi.h"
+#include "esp_wifi_default.h"
+#include "esp_err.h"
+#include "esp_netif.h"
+#include "esp_sntp.h"
+#include "esp_log.h"
+#include "freertos/FreeRTOS.h"
+#include "freertos/task.h"
+#include "freertos/semphr.h"
+#include "nvs_flash.h"
+
+#define SNTP_SERVER_FQDN "pool.ntp.org"
+
+static const char *TAG = "sample_azureiot";
+
+static bool g_timeInitialized = false;
+
+/*-----------------------------------------------------------*/
+
+extern void vStartDemoTask( void );
+
+/*-----------------------------------------------------------*/
+
+void vLoggingPrintf( const char * pcFormat,
+                     ... )
+{
+    va_list arg;
+
+    va_start( arg, pcFormat );
+    vprintf( pcFormat, arg );
+    va_end( arg );
+}
+
+/*-----------------------------------------------------------*/
+
+uint64_t ullGetUnixTime( void )
+{
+    time_t now = time(NULL);
+
+    if (now == (time_t)(-1))
+    {
+        ESP_LOGE(TAG, "Failed obtaining current time.\r\n");
+    }
+
+    return now;
+}
+
+/*-----------------------------------------------------------*/
+
+static xSemaphoreHandle s_semph_get_ip_addrs;
+static esp_ip4_addr_t s_ip_addr;
+
+#define NR_OF_IP_ADDRESSES_TO_WAIT_FOR 1
+
+#if CONFIG_SAMPLE_IOT_WIFI_SCAN_METHOD_FAST
+#define SAMPLE_IOT_WIFI_SCAN_METHOD WIFI_FAST_SCAN
+#elif CONFIG_SAMPLE_IOT_WIFI_SCAN_METHOD_ALL_CHANNEL
+#define SAMPLE_IOT_WIFI_SCAN_METHOD WIFI_ALL_CHANNEL_SCAN
+#endif
+
+#if CONFIG_SAMPLE_IOT_WIFI_CONNECT_AP_BY_SIGNAL
+#define SAMPLE_IOT_WIFI_CONNECT_AP_SORT_METHOD WIFI_CONNECT_AP_BY_SIGNAL
+#elif CONFIG_SAMPLE_IOT_WIFI_CONNECT_AP_BY_SECURITY
+#define SAMPLE_IOT_WIFI_CONNECT_AP_SORT_METHOD WIFI_CONNECT_AP_BY_SECURITY
+#endif
+
+#if CONFIG_SAMPLE_IOT_WIFI_AUTH_OPEN
+#define SAMPLE_IOT_WIFI_SCAN_AUTH_MODE_THRESHOLD WIFI_AUTH_OPEN
+#elif CONFIG_SAMPLE_IOT_WIFI_AUTH_WEP
+#define SAMPLE_IOT_WIFI_SCAN_AUTH_MODE_THRESHOLD WIFI_AUTH_WEP
+#elif CONFIG_SAMPLE_IOT_WIFI_AUTH_WPA_PSK
+#define SAMPLE_IOT_WIFI_SCAN_AUTH_MODE_THRESHOLD WIFI_AUTH_WPA_PSK
+#elif CONFIG_SAMPLE_IOT_WIFI_AUTH_WPA2_PSK
+#define SAMPLE_IOT_WIFI_SCAN_AUTH_MODE_THRESHOLD WIFI_AUTH_WPA2_PSK
+#elif CONFIG_SAMPLE_IOT_WIFI_AUTH_WPA_WPA2_PSK
+#define SAMPLE_IOT_WIFI_SCAN_AUTH_MODE_THRESHOLD WIFI_AUTH_WPA_WPA2_PSK
+#elif CONFIG_SAMPLE_IOT_WIFI_AUTH_WPA2_ENTERPRISE
+#define SAMPLE_IOT_WIFI_SCAN_AUTH_MODE_THRESHOLD WIFI_AUTH_WPA2_ENTERPRISE
+#elif CONFIG_SAMPLE_IOT_WIFI_AUTH_WPA3_PSK
+#define SAMPLE_IOT_WIFI_SCAN_AUTH_MODE_THRESHOLD WIFI_AUTH_WPA3_PSK
+#elif CONFIG_SAMPLE_IOT_WIFI_AUTH_WPA2_WPA3_PSK
+#define SAMPLE_IOT_WIFI_SCAN_AUTH_MODE_THRESHOLD WIFI_AUTH_WPA2_WPA3_PSK
+#elif CONFIG_SAMPLE_IOT_WIFI_AUTH_WAPI_PSK
+#define SAMPLE_IOT_WIFI_SCAN_AUTH_MODE_THRESHOLD WIFI_AUTH_WAPI_PSK
+#endif
```

Comment:
can we move this to before functions. like before line 33?

also please make sure :

All:
#define 

All:
global data allocations

All:
static functions

All:
non-static function

Reply:
Done.